backend/app/app.js

```diff
@@ -44,31 +42,40 @@ var app = express();
 app.use(bodyParser.json());
 app.use(bodyParser.urlencoded({extended: false}));
 app.use(cookieParser());
+app.use(session({secret: "greenpastures", resave: false, saveUninitialized: false}));
 app.use(passport.initialize());
 app.use(passport.session());
 app.use(logger("dev"));
+app.use(express.static(path.join(__dirname, "views")));
 
 // Routes
 var index = require("./routes/index");
-// var api = require('./routes/api');
 
-// Register static files
-app.use(express.static(path.join(__dirname, "views")));
-
-// Route Usage
 app.use("/", index);
-// app.use('/api', api) : in-progress
-
-app.post("/login", passport.authenticate("local"));
 
+app.post("/login", passport.authenticate("local", {failureRedirect: "/login"}), function(req, res) {
+    console.log("User " + req.body.username + "has logged in.");
+    res.send("ok");
+});
 
-app.use("/users", function(req, res) {
+app.get("/users", isLoggedIn, function(req, res) {
     User.find()
         .exec(function(err, users) {
             if (!err) { res.json(users); }
         });
 });
 
+app.get("/logout", function(req, res) {
```

Comment:
logging out should be a post request in my opinion.

Reply:
I used "get" because we doesn't seem to passing data to the request. unless you want something to be passed to the server. what do you have in mind?

Reply:
Well but also you're not "getting" stuff too. :) That's just a question of convention anyway. Judging by this article they seem to not agree with it as well

https://stackoverflow.com/questions/3521290/logout-get-or-post

Reply:
well that is one point. haha. i'll read that later thanks :)